src/pnm/encoder.rs

```diff
@@ -237,10 +238,10 @@ impl<W: Write> PNMEncoder<W> {
                 encoded: None,
             },
             (_, _) => {
-                return Err(io::Error::new(
+                return Err(ImageError::IoError(io::Error::new(
```

Comment:
Here a FIXME note would be appreciated if the line is touched anyways. Maybe `io::Error` is not the worst here since the request was invalid, not unsupported.